### PR TITLE
Added new section synclogs to show the aggregated reports history

### LINKED ISSUE #12: Introduce an aggregated reports log
Currently, the app is deleting a data entry each time it is sent to the server. So the user doesn't have any way to know whether his data was or not sent to the server.

We should:
1) Create a new section in the drawer menu that slides from left clicking on the menu icon saying "Sync logs"
2) Each time an aggregated report is sent, store a line in a log file
3) Create a new screen showing the log content with a "Reset" button to delete the log (in order to avoid it to take all the space in disc at some point)

### PR DESCRIPTION
**Pull request by idelcano**
Closes #12

### CODE REVIEW

**Comment by idelcano** [comment]
@xurxodev  fixed!

File: dhis2-android-app/src/main/java/org/dhis2/mobile/processors/OfflineDataProcessor.java [hunk 1]
```diff
@@ -107,7 +111,10 @@ private static void uploadOfflineReports(Context context) {
                         description = ImportSummariesHandler.getDescription(resp.getBody(),
                                 context.getString(R.string.import_failed));
                     }
-                    String message = String.format("(%s) %s", info.getPeriodLabel(), info.getFormLabel());
+
```
Comment (xurxodev) [inline dhis2-android-app/src/main/java/org/dhis2/mobile/processors/OfflineDataProcessor.java]: @idelcano this code for log mixed with other logic is a little dirty, can you extract a separate method or class?

File: dhis2-android-app/src/main/java/org/dhis2/mobile/processors/OfflineDataProcessor.java [hunk 2]
```diff
@@ -116,6 +123,14 @@ private static void uploadOfflineReports(Context context) {
                     // Removing uploaded data
                     TextFileUtils.removeFile(reportFile);
                     PrefUtils.removeOfflineReportInfo(context, reportFile.getName());
+                } else {
+                    message = message + context.getString(R.string.network_error) + ": "
```
Comment (xurxodev) [inline dhis2-android-app/src/main/java/org/dhis2/mobile/processors/OfflineDataProcessor.java]: @idelcano this code for log mixed with other logic is a little dirty, can you extract a separate method or class?

File: dhis2-android-app/src/main/java/org/dhis2/mobile/processors/ReportUploadProcessor.java [hunk 3]
```diff
@@ -87,10 +88,22 @@ public static void upload(Context context, DatasetInfoHolder info, ArrayList<Gro
                         context.getString(R.string.import_failed));
             }
 
+            message = message + description;
```
Comment (xurxodev) [inline dhis2-android-app/src/main/java/org/dhis2/mobile/processors/ReportUploadProcessor.java]: @idelcano this code for log mixed with other logic is a little dirty, can you extract a separate method or class?

File: dhis2-android-app/src/main/java/org/dhis2/mobile/processors/ReportUploadProcessor.java [hunk 4]
```diff
@@ -87,10 +88,22 @@ public static void upload(Context context, DatasetInfoHolder info, ArrayList<Gro
...
+            TextFileUtils.saveLogMessage(context, message);
+
             String title = description;
-            String message = String.format("(%s) %s", info.getPeriodLabel(), info.getFormLabel());
+            message = String.format("(%s) %s", info.getPeriodLabel(), info.getFormLabel());
             NotificationBuilder.fireNotification(context, title, message);
+
+
         } else {
+            message = message + context.getString(R.string.network_error) + ": "
```
Comment (xurxodev) [inline dhis2-android-app/src/main/java/org/dhis2/mobile/processors/ReportUploadProcessor.java]: @idelcano this code for log mixed with other logic is a little dirty, can you extract a separate method or class?

File: dhis2-android-app/src/main/java/org/dhis2/mobile/utils/TextFileUtils.java [hunk 5]
```diff
@@ -37,18 +37,30 @@
 import java.io.FileOutputStream;
 import java.io.FileReader;
 import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 
 public final class TextFileUtils {
     private static final String TAG = TextFileUtils.class.getSimpleName();
 
+    public static void saveLogMessage(Context context, String message) {
```
Comment (xurxodev) [inline dhis2-android-app/src/main/java/org/dhis2/mobile/utils/TextFileUtils.java]: @idelcano this method not use existed standards for this file, can you refactor to for create a overload for WriteTextFile, for example:
writeTextFile(Context context, Directory dir, FileNames name, String data, includeDate)

File: dhis2-android-app/src/main/java/org/dhis2/mobile/utils/TextFileUtils.java [hunk 6]
```diff
@@ -115,6 +127,32 @@ public static void writeTextFile(Context context, Directory dir, FileNames name,
         writeTextFile(context, dir, name.toString(), data);
     }
 
+
+
+    public static void writeLineIntoTextFile(Context context, Directory dir,
```
Comment (xurxodev) [inline dhis2-android-app/src/main/java/org/dhis2/mobile/utils/TextFileUtils.java]: this code has duplicate code with writeTextFile, can you refactor writeTextFile according to parameter for add formatted formattedDate after text content to add and remove this method?

**Comment by idelcano** [comment]
@xurxodev changes done

**Pull request merged**